src/nupic/algorithms/Connections.cpp

```diff
@@ -51,6 +66,13 @@ void Connections::initialize(CellIdx numCells) {
   nextEventToken_ = 0;
 }
 
+void Connections::initialize(CellIdx numCells, Permanence connectedThreshold) {
```

Comment:
what is the motivation for almost duplicit constructors, initialize() for (numCells), and (numCells, connectedThreshold)? 
If there is, can we unify? Maybe `Connections(UInt numCells, Permanence connectedThreshold = 0.2f)` ?